MapboxNavigation/NavigationMapView.swift

```diff
@@ -1025,39 +1027,122 @@ extension Dictionary where Key == Int, Value: MGLStyleValue<NSNumber> {
 }
 
 //MARK: NavigationMapViewDelegate
-
+/**
+ The `NavigationMapViewDelegate` provides methods for configuring the NavigationMapView, as well as responding to events triggered by the NavigationMapView.
+ */
 @objc(MBNavigationMapViewDelegate)
 public protocol NavigationMapViewDelegate: class  {
+    /**
+     Asks the reciever to return an MGLStyleLayer for routes, given an identifier and source.
+     - note: This method is invoked at initalization time and anytime routes are added.
+     - parameter mapView: the NavigationMapView.
+     - parameter identifier: the style identifier.
+     - parameter source: The Layer source provided.
+     - returns: an MGLStyleLayer that the map applies to all routes.
+     */
     @objc optional func navigationMapView(_ mapView: NavigationMapView, routeStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
+    /**
+     Asks the reciever to return an MGLStyleLayer for waypoints, given an identifier and source.
+     - note: This method is invoked at initalization time and anytime waypoints are added.
+     - parameter mapView: the NavigationMapView.
+     - parameter identifier: the style identifier.
+     - parameter source: The Layer source provided.
+     - returns: an MGLStyleLayer that the map applies to all waypoints.
+     */
     @objc optional func navigationMapView(_ mapView: NavigationMapView, waypointStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
+    /**
+     Asks the reciever to return an MGLStyleLayer for waypoint symbols, given an identifier and source.
+     - note: This method is invoked at initalization time and anytime waypoints are added.
+     - parameter mapView: the NavigationMapView.
+     - parameter identifier: the style identifier.
+     - parameter source: The Layer source provided.
+     - returns: an MGLStyleLayer that the map applies to all waypoint symbols.
+     */
     @objc optional func navigationMapView(_ mapView: NavigationMapView, waypointSymbolStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
+    /**
+     Asks the reciever to return an MGLStyleLayer for route casings, given an identifier and source.
+     - note: This method is invoked at initalization time and anytime routes are added.
```

Comment:
Along with this second paragraph, say something about why one might want to set a route casing. For example:

> Specify a casing to ensure good contrast between the route line and the underlying map layers.